### PR TITLE
To-do list code review

### PR DESCRIPTION
**Pull request by nedith**
### Implemented Features:
- Implemented a function for adding a new task (add a new element to the array).
- Implemented a function for deleting a task (remove an element from the array).
- Implemented a function for editing task descriptions.
- By default new tasks have the property completed set to false and the property index set to the value of the new array length
- Deleting a task updates all remaining items' indexes, so they represent the current list order and are unique
- All changes to the To-Do List are saved in local storage.
#### Please let me know what I need to improve :eyeglasses:

### CODE REVIEW

File: src/modules/crud.js [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+  const newListItem = new Task(newDescription, false, checkbox.length);
+  taskArray.push(newListItem);
+  const stringedItems = JSON.stringify(taskArray);
+  localStorage.setItem('items', stringedItems);
+  // Edit task
+  const editItem = document.querySelectorAll('.ellipsis');
+  editItem.forEach((item) => {
+    item.addEventListener('click', () => {
+      item.parentNode.classList.add('clicked-on');
+      editItems(item.previousElementSibling);
```
Comment (saqibjavaiddev) [inline src/modules/crud.js:94]: You are violating the **DRY** principle which is not to repeat your code. Kindly check you are repeating the same code two times. If you have any doubt you may check out this article [DRY, KISS and YAGNI](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/clean-code/lesson_DRY_KISS_YAGNI.md)

File: src/modules/crud.js [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+  <span class="trashcan">
+  <i id="trash${item.index}" class="fa-solid fa-trash-can"></i>
+  </span>
+  `;
+    listOfTasks.appendChild(newItemList);
+    const editItem = document.querySelectorAll('.ellipsis');
+    editItem.forEach((item) => {
+      item.addEventListener('click', () => {
+        item.parentNode.classList.add('clicked-on');
+        editItems(item.previousElementSibling, newItemList);
```
Comment (saqibjavaiddev) [inline src/modules/crud.js:145]: You are repeating the same code, I suggest to try creating a function and then reuse the code

**Review by saqibjavaiddev — CHANGES_REQUESTED** [review]
## Hi @nedith,

**Good job so far! 👍🏻**
There is an issue that you can fix, but you are almost there!💪


## To Highlight

✅ Following best JS coding practices
✅ No JS Linter Errors
✅ Using ES6 functionality

## Required Changes ♻️

 _Check the comments under the review for more changes_.

File: src/modules/crud.js (hunk 2 above)
Comment (nedith) [reply src/modules/crud.js]: Thank you for the review:smiley: I will refactor accordingly :muscle:

**Comment by saqibjavaiddev** [comment]
## STATUS: APPROVED ✔️ ✔️

Hi @nedith ,

✔️ Using Best JS Practices

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

**Review by saqibjavaiddev — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @nedith ,

✔️ Using Best JS Practices

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

**Pull request merged**